src/utils/memory.hpp

```diff
@@ -469,7 +469,8 @@ class PoolResource final : public MemoryResource {
   /// impl::Pool::MaxBlocksInChunk()) as the real maximum number of blocks per
   /// chunk. Allocation requests exceeding max_block_size are simply forwarded
   /// to upstream memory.
-  PoolResource(size_t max_blocks_per_chunk, size_t max_block_size, MemoryResource *memory = NewDeleteResource());
+  PoolResource(size_t max_blocks_per_chunk, size_t max_block_size, MemoryResource *memory_pools = NewDeleteResource(),
+               MemoryResource *memory_unpooled = NewDeleteResource());
```

Comment:
Having a default value is okay, however in the `PullPlan::Pull` I would explicitly pass in a `NewDeleteResource` to express the intent: we want all of the big blocks are allocated with `NewDeleteResource`.

Reply:
Makes sense! Added that part.